Various changes

kobuki opened the pull request:
The changes include the following. Please discuss with Luc first if you feel you need to. I've been in direct contact with him regarding almost all important changes.
- changed system to US, since consistently provides better wind values, with and without error corrections - the rest of the metrics are properly converted to US on loop packet output
- added wind error correction using the new table - pretty good with the angles below 180° but will need to do the correction table above 180° too because corrections don't look symmetric WRT E/W
- added support for the RH output of the custom receiver I've made for Frank (and for anyone else, code soon to be published)

kobuki committed:
Small fixes and wind EC test 1
- fixed detection of missing solar sensor (Luc)
- added wind value error correction using an old Davis correction table and linear interpolation (not too useful, but adding here to have it in history)

kobuki committed:
Added new wind error correction method
- added new wind error correction table and method using logged data
- fixed Pro wind direction

kobuki committed:
Small fix

kobuki committed:
Fixed ugly copy-paste error in interpolate()

kobuki committed:
Added some debug code

kobuki committed:
Wind testing commit, not for general use

kobuki committed:
Several changes
- fixed remaining US values
- added inHumidity to B packet for the new custom receiver

kobuki committed:
Fixed inHumidity float

kobuki committed:
Cosmetics

matthewwall commented:
a few things:
- for the "Small fixes and wind EC test 1" commit i get "1,334 additions, 1,229 deletions not shown because the diff is too large. Please use a local Git client to view these changes. "  really?  so it looks like that is predominantly ec code, but a solar radiation fix sneaked in there too.
- the default sensor map should not include debug mappings.  put those (such as windSpeedRaw) in your own weewx.conf, since no one else will have that schema anyway.
- use weewx.formulas.CtoF instead of pulling in all of the unit conversion overhead
- same for pressure conversion - just multiply by wxformulas.INHG_PER_MBAR
- do not calculate rain_rate in the driver.  using a direct expansion of bucket tips to rainfall-in-an-hour is going to be nowhere close to davis' algorithm.  and matching davis' algorithm for rain rate does not belong in the driver anyway.
- why is there a "not for general use" commit in here?
- exactly what is this error-corrected wind speed, anyway?  the driver should report what it gets from the sensors.  if someone wants to apply error correction or any other kind of massaging of the data, they should do that in StdCalibrate.  that is the "official weewx" stance, anyway.  is the meteostick output so bad that it needs correction?  is the correction exactly the davis correction, or is it a correction that makes the meteostick output close to davis corrected output?  (some higher-level description of purpose in the code would be nice)
- document the wind correction experiments in separate branches in the repository, not in revisions of the master (that also makes it easier to run experiments that compare the different error correction algorithms).

kobuki commented:
All in order:
- Please point out what you don't understand so I can explain. The "sneak" was a small fix I didn't document separately, I didn't think it warrants a mention - it should have been included a lot earlier.
- Without the map entry I can't put it in my local conf but I'll remove it. It's not needed any more.
- CtoF: All magical constants don't exist in the core. But they DO exist in the unit conversion code. So instead of making up our own magical constants and trying to find the existing ones, it's much more elegant to use the existing converter. Nothing much of an overhead. For the core it should have been loaded already in memory.
- Pressure: same as above.
- Rain rate: I changed the necessary constant from mm to inches since the driver now works with US units, nothing more. The rest is courtesy of Luc, I didn't touch anything else. Please discuss with him - although in our private list you seem to have been involved and understood the changes he made.
- Not for general use because it was a test version. I should have put it in a separate branch, I admint that. I didn't intend to make a pull request then, but I think there are very important changes so I finally did.
- Wind EC: we've been discussing this on the private list back and forth. In short: The raw values from the anemometer are of no use. The console corrects them using a lookup table which we do not have. I've logged raw vs. corrected values from my Envoy and made a table. I can explain it in detail if you want, please drop me a mail of you want me to do so. It belongs to the driver since without this the core is not receiving any useful wind speed values. Yes, it's that bad.
- Well, you know, I don't really want to spend much time very deeply documenting this but I'd like to go this code in the main branch so we get useful wind values.

I'll drop you a mail to discuss on how I should proceed. Until then I continue to commit to my current repo.

kobuki commented:
Per our discussion, I'll fork your repo anew (managing it in a separate fork is tedious with GitHub, sorry) and close this pull request.
